Write add_scalars data in the same file

nataliakliushkina opened the pull request:
Added multifile flag to write scalar data into separate files. This can slow down dashboard loading.

orionr commented on torch/utils/tensorboard/writer.py:
```diff
@@ -292,7 +292,8 @@ def add_scalar(self, tag, scalar_value, global_step=None, walltime=None):
 
-    def add_scalars(self, main_tag, tag_scalar_dict, global_step=None, walltime=None):
+    def add_scalars(self, main_tag, tag_scalar_dict, global_step=None,
+                    walltime=None):
```
Nit - can you move the `walltime=None` back to the previous line for no changes here?

orionr reviewed (approved):
Looks good! One nit request and let's make sure all the tests pass before landing.

nataliakliushkina committed:
Write add_scalars data in the same file (#21100)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/21100

Added multifile flag to write scalar data into separate files. This can slow down dashboard loading.

Reviewed By: orionr

Differential Revision: D15548913

fbshipit-source-id: 9ae33122392bb728d097ea6dd83014e3c9754df7

sw-gong commented:
Why this change?

The same amount of log files (with each one corresponding to a specific scalar) will be created before or after this commit; however, this commit corrupted the initial well-organised and easy-to-ready directory structure. 

Let's consider a scenario where one attempts to log multiple runs (with varied train/val/test split) altogether with train/val/test losses as well as train/val/test accuracies. 
`writer.add_scalars('{run}_run', {'train_loss': train_loss, 'val_loss': val_loss, 'test_loss': test_loss}, epoch)`
`writer.add_scalars('{run}_run', {'train_loss': train_loss, 'val_loss': val_loss, 'test_loss': test_loss}, epoch)`

All the subfolders related to each scalar will be located at the same level, e.g., 
`/main_folder/1_run_train_loss_`, 
`/main_folder/1_run_val_loss_`
`/main_folder/1_run_test_loss_`
`/main_folder/2_run_train_loss_`
....
`/main_folder/1_run_train_acc`
...

Whereas before this commit (and still the practice in TensorboardX), e.g.,
'/main_folder/1_run/train_loss`,
'/main_folder/1_run/val_loss`,
'/main_folder/1_run/test_loss`,
'/main_folder/1_run/train_acc`,
'/main_folder/1_run/val_acc`,
'/main_folder/1_run/test_acc`,
this is much clearer, and importantly, different scalars will always be written in different log file.
